File: sol/1287/1287.cpp
struct Solution {
    int findSpecialInteger(const vector<int>& arr) {
        int lastEle = -1, combo = 0;
        for (const int &n : arr){
            if (n == lastEle){
                ++combo;
            } else {
                combo = 1;
            }
            lastEle = n;
            if (combo * 4 > arr.size()){
                break;
            }
        }
        return lastEle;
    }
};

File: sol/1287/1287.py
class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        n = len(arr)
        lastEle = None
        combo = 0
        for ele in arr:
            if ele == lastEle:
                combo += 1
            else:
                combo = 1
            lastEle = ele
            if combo * 4 > n:
                return ele
